src/Connection.php

```diff
@@ -370,9 +370,14 @@ private function getCriteriaCondition(array $criteria): array
                 continue;
             }
 
-            $columns[]    = $columnName;
-            $values[]     = $value;
-            $conditions[] = $columnName . ' = ?';
+            if (is_array($value)) {
```

Comment:
We cannot guess that an `IN` query must be used based on the value being an array. This would break types that map to a PHP array like `JSON` for instance.